From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Wed, 10 Apr 2024 13:51:33 +0000
Subject: [PATCH] config-parser: Add support for conditional parsing of
 portions of config app

This will be needed for conditional parsing of content filtering files
based on user preference on language settings.
---
 vanadium/android_config/proto/BUILD.gn          |  1 +
 .../config/VanadiumConfConditionals.java        | 17 +++++++++++++++++
 2 files changed, 18 insertions(+)
 create mode 100644 vanadium/android_config/proto/java/src/app/vanadium/config/VanadiumConfConditionals.java

diff --git a/vanadium/android_config/proto/BUILD.gn b/vanadium/android_config/proto/BUILD.gn
index 31aef8a25d16a..5f04b2dca6267 100644
--- a/vanadium/android_config/proto/BUILD.gn
+++ b/vanadium/android_config/proto/BUILD.gn
@@ -54,6 +54,7 @@ action("config_generation") {
 
 android_library("browser_config_parser_java") {
   sources = [
+    "java/src/app/vanadium/config/VanadiumConfConditionals.java",
     "java/src/app/vanadium/config/VanadiumConfParser.java",
   ]
   deps = [
diff --git a/vanadium/android_config/proto/java/src/app/vanadium/config/VanadiumConfConditionals.java b/vanadium/android_config/proto/java/src/app/vanadium/config/VanadiumConfConditionals.java
new file mode 100644
index 0000000000000..ec1be4d92d378
--- /dev/null
+++ b/vanadium/android_config/proto/java/src/app/vanadium/config/VanadiumConfConditionals.java
@@ -0,0 +1,17 @@
+package app.vanadium.config;
+
+public final class VanadiumConfConditionals {
+
+    public interface ConditionSupplier {
+    }
+
+    private static volatile ConditionSupplier conditionSupplier;
+
+    private static ConditionSupplier getConditionSupplier() {
+        return VanadiumConfConditionals.conditionSupplier;
+    }
+
+    public static void setBrowserConditionals(ConditionSupplier conditionSupplier) {
+        VanadiumConfConditionals.conditionSupplier = conditionSupplier;
+    }
+}
